addons/calendar_sms/models/calendar_alarm_manager.py

```diff
@@ -16,9 +16,9 @@ def _send_reminder(self):
         if not events_by_alarm:
             return
 
-        event_ids = list(set(event_id for event_ids in events_by_alarm.values() for event_id in event_ids))
-        events = self.env['calendar.event'].browse(event_ids)
         alarms = self.env['calendar.alarm'].browse(events_by_alarm.keys())
-        for event in events:
-            alarm = event.alarm_ids.filtered(lambda alarm: alarm.id in alarms.ids)
-            event._do_sms_reminder(alarm)
+        for alarm in alarms:
+            event_ids = events_by_alarm.get(alarm.id, [])
+            events = self.env['calendar.event'].browse(event_ids)
+            for event in events:
+                event._do_sms_reminder(alarm)
```

Comment:
If you browse them in a loop it means we'll potentially have to do a query at every stage of the loop.

Not a bad idea to browse instead of filtering but you should add prefetch ids (`with_prefetch`)